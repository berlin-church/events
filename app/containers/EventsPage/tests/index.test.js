/**
 *
 * Tests for EventsPage
 *
 * @see https://github.com/react-boilerplate/react-boilerplate/tree/master/docs/testing
 *
 */

import React from 'react';
import { render } from 'react-testing-library';
import { IntlProvider } from 'react-intl';
// import 'jest-dom/extend-expect'; // add some helpful assertions

import { Provider } from 'react-redux';
import configureStore from '../../../configureStore';

import history from '../../../utils/history';

import { EventsPage } from '../index';
import { DEFAULT_LOCALE } from '../../../i18n';

const store = configureStore({}, history);

describe('<EventsPage />', () => {
  it('Expect to not log errors in console', () => {
    const spy = jest.spyOn(global.console, 'error');
    const dispatch = jest.fn();
    render(
      <Provider store={store}>
        <IntlProvider locale={DEFAULT_LOCALE}>
          <EventsPage dispatch={dispatch} />
        </IntlProvider>
      </Provider>,
    );
    expect(spy).not.toHaveBeenCalled();
  });

  it('Should render and match the snapshot', () => {
    const {
      container: { firstChild },
    } = render(
      <Provider store={store}>
        <IntlProvider locale={DEFAULT_LOCALE}>
          <EventsPage />
        </IntlProvider>
      </Provider>,
    );
    expect(firstChild).toMatchSnapshot();
  });
});
