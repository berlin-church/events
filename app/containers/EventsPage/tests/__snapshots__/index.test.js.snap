// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<EventsPage /> Should render and match the snapshot 1`] = `
<div>
  <div
    class="makeStyles-root-88"
  >
    <header
      class="MuiPaper-root MuiPaper-elevation4 MuiAppBar-root MuiAppBar-positionStatic MuiAppBar-colorPrimary makeStyles-appBar-89"
    >
      <div
        class="MuiToolbar-root MuiToolbar-regular MuiToolbar-gutters makeStyles-title-90"
      >
        <img
          alt="Berlin International Community Church"
          class="makeStyles-logo-91"
          src="//static1.squarespace.com/static/589397498419c2f6476f2c77/t/5b28efa970a6ad0778246fc3/1539028385210/?format=1500w"
        />
      </div>
    </header>
  </div>
  <div
    class="EventCardsList__Wrapper-k4npy7-0 kgSFEk"
  >
    <div
      class="MuiPaper-root MuiPaper-elevation1 MuiPaper-rounded MuiCard-root makeStyles-card-132"
    >
      <button
        class="MuiButtonBase-root MuiCardActionArea-root"
        tabindex="0"
        type="button"
      >
        <img
          class="MuiCardMedia-root MuiCardMedia-media makeStyles-media-133"
          src="https://res.cloudinary.com/dkbwcwx1s/image/upload/v1536004712/connect.jpg"
          title="connect"
        />
        <div
          class="makeStyles-title-135"
        >
          <span>
            CONNECT
          </span>
        </div>
        <div
          class="MuiCardContent-root"
        />
        <span
          class="MuiCardActionArea-focusHighlight"
        />
        <span
          class="MuiTouchRipple-root"
        />
      </button>
      <div
        class="MuiCardActions-root MuiCardActions-spacing"
      >
        <button
          class="MuiButtonBase-root MuiButton-root makeStyles-link-134 MuiButton-text MuiButton-sizeSmall"
          tabindex="0"
          type="button"
        >
          <span
            class="MuiButton-label"
          >
            register
          </span>
          <span
            class="MuiTouchRipple-root"
          />
        </button>
      </div>
    </div>
    <div
      class="MuiPaper-root MuiPaper-elevation1 MuiPaper-rounded MuiCard-root makeStyles-card-132"
    >
      <button
        class="MuiButtonBase-root MuiCardActionArea-root"
        tabindex="0"
        type="button"
      >
        <img
          class="MuiCardMedia-root MuiCardMedia-media makeStyles-media-133"
          src="https://res.cloudinary.com/dkbwcwx1s/image/upload/v1528827293/placeholder.png"
          title="host a community group"
        />
        <div
          class="makeStyles-title-135"
        >
          <span>
            HOST A COMMUNITY GROUP
          </span>
        </div>
        <div
          class="MuiCardContent-root"
        />
        <span
          class="MuiCardActionArea-focusHighlight"
        />
        <span
          class="MuiTouchRipple-root"
        />
      </button>
      <div
        class="MuiCardActions-root MuiCardActions-spacing"
      >
        <button
          class="MuiButtonBase-root MuiButton-root makeStyles-link-134 MuiButton-text MuiButton-sizeSmall"
          tabindex="0"
          type="button"
        >
          <span
            class="MuiButton-label"
          >
            register
          </span>
          <span
            class="MuiTouchRipple-root"
          />
        </button>
      </div>
    </div>
    <div
      class="MuiPaper-root MuiPaper-elevation1 MuiPaper-rounded MuiCard-root makeStyles-card-132"
    >
      <button
        class="MuiButtonBase-root MuiCardActionArea-root"
        tabindex="0"
        type="button"
      >
        <img
          class="MuiCardMedia-root MuiCardMedia-media makeStyles-media-133"
          src="https://res.cloudinary.com/dkbwcwx1s/image/upload/v1536004711/CG/community-group.jpg"
          title="community group"
        />
        <div
          class="makeStyles-title-135"
        >
          <span>
            COMMUNITY GROUP
          </span>
        </div>
        <div
          class="MuiCardContent-root"
        />
        <span
          class="MuiCardActionArea-focusHighlight"
        />
        <span
          class="MuiTouchRipple-root"
        />
      </button>
      <div
        class="MuiCardActions-root MuiCardActions-spacing"
      >
        <button
          class="MuiButtonBase-root MuiButton-root makeStyles-link-134 MuiButton-text MuiButton-sizeSmall"
          tabindex="0"
          type="button"
        >
          <span
            class="MuiButton-label"
          >
            see all
          </span>
          <span
            class="MuiTouchRipple-root"
          />
        </button>
      </div>
    </div>
  </div>
</div>
`;
